exporters/autoexport/metrics.go

```diff
@@ -0,0 +1,102 @@
+	"go.opentelemetry.io/otel/sdk/metric"
+)
+
+// MetricOption applies an autoexport configuration option.
+type MetricOption = option[metric.Reader]
+
+// WithFallbackMetricReader sets the fallback exporter to use when no exporter
+// is configured through the OTEL_METRICS_EXPORTER environment variable.
+func WithFallbackMetricReader(exporter metric.Reader) MetricOption {
+	return withFallback[metric.Reader](exporter)
+}
+
+// NewMetricReader returns a configured [go.opentelemetry.io/otel/sdk/metric.Reader]
+// defined using the environment variables described below.
+//
+// OTEL_METRICS_EXPORTER defines the metrics exporter; supported values:
+//   - "none" - "no operation" exporter
+//   - "otlp" (default) - OTLP exporter; see [go.opentelemetry.io/otel/exporters/otlp/otlpmetric]
+//   - "prometheus" - Prometheus exporter; see [go.opentelemetry.io/otel/exporters/prometheus]
+//
+// OTEL_EXPORTER_OTLP_PROTOCOL defines OTLP exporter's transport protocol;
+// supported values:
+//   - "grpc" - protobuf-encoded data using gRPC wire format over HTTP/2 connection;
+//     see: [go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc]
+//   - "http/protobuf" (default) -  protobuf-encoded data over HTTP connection;
+//     see: [go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp]
+//
+// An error is returned if an environment value is set to an unhandled value.
+//
+// Use [RegisterMetricReader] to handle more values of OTEL_METRICS_EXPORTER.
+//
+// Use [WithFallbackMetricReader] option to change the returned exporter
+// when OTEL_TRACES_EXPORTER is unset or empty.
+//
+// Use [IsNoneMetricReader] to check if the retured exporter is a "no operation" exporter.
+func NewMetricReader(ctx context.Context, opts ...MetricOption) (metric.Reader, error) {
+	return metricsSignal.create(ctx, opts...)
+}
+
+// RegisterMetricReader sets the MetricReader factory to be used when the
+// OTEL_METRICS_EXPORTERS environment variable contains the exporter name. This
+// will panic if name has already been registered.
+func RegisterMetricReader(name string, factory func(context.Context) (metric.Reader, error)) {
+	must(metricsSignal.registry.store(name, factory))
+}
+
+var metricsSignal = newSignal[metric.Reader]("OTEL_METRICS_EXPORTER")
+
+func init() {
+	RegisterMetricReader("otlp", func(ctx context.Context) (metric.Reader, error) {
+		proto := os.Getenv(otelExporterOTLPProtoEnvKey)
+		if proto == "" {
+			proto = "http/protobuf"
+		}
+
+		switch proto {
+		case "grpc":
+			r, err := otlpmetricgrpc.New(ctx)
+			if err != nil {
+				return nil, err
+			}
+			return metric.NewPeriodicReader(r), nil
+		case "http/protobuf":
+			r, err := otlpmetrichttp.New(ctx)
+			if err != nil {
+				return nil, err
+			}
+			return metric.NewPeriodicReader(r), nil
+		default:
+			return nil, errInvalidOTLPProtocol
+		}
+	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		return prometheus.New()
```

Comment:
**Non-blocking for this PR**:  Note that this does not actually serve the prometheus metric endpoint.  A user would still need to create the prometheus handler and listen on a port.

This could be a reasonable place to implement the prometheus exporter environment variables: https://github.com/open-telemetry/opentelemetry-specification/blob/563958cb2bd8529990f19fdce7a5f3643bf63091/specification/configuration/sdk-environment-variables.md?plain=1#L201

Reply:
Should they be implemented in this package or in the Prometheus package?

Reply:
I would rather implement it in this package.

Reply:
Sounds good, I have created #4472 to track that work.